egui_demo_lib/src/apps/demo/widgets.rs

```diff
@@ -133,7 +152,26 @@ impl Widgets {
             }
         });
 
+        ui.separator();
+
         ui.label("Multiline text input:");
         ui.text_edit_multiline(&mut self.multiline_text_input);
+
+        ui.separator();
+
+        ui.horizontal(|ui| {
+            ui.label("Code editor:");
+
+            ui.separator();
+
+            ui.checkbox(&mut self.tab_as_spaces, "Tabs as spaces");
+            ui.checkbox(&mut self.tab_moves_focus, "Tabs moves focus");
```

Comment:
```suggestion
            ui.checkbox(&mut self.tab_moves_focus, "Tabs moves focus");
            if !self.tab_moves_focus {
                ui.checkbox(&mut self.tab_as_spaces, "Tabs as spaces");
            }
```

(the `tab_as_spaces` is meaningless if tab moves focus)